File: src/main/java/no/nav/sosialhjelp/soknad/business/soknadunderbehandling/SoknadUnderArbeidRepository.java
package no.nav.sosialhjelp.soknad.business.soknadunderbehandling;

import no.nav.sosialhjelp.soknad.business.exceptions.SamtidigOppdateringException;
import no.nav.sosialhjelp.soknad.domain.SoknadUnderArbeid;

import java.util.List;
import java.util.Optional;

public interface SoknadUnderArbeidRepository {

    Long opprettSoknad(SoknadUnderArbeid soknadUnderArbeid, String eier);
    Optional<SoknadUnderArbeid> hentSoknad(Long soknadId, String eier);
    SoknadUnderArbeid hentSoknad(String behandlingsId, String eier);
    Optional<SoknadUnderArbeid> hentSoknadOptional(String behandlingsId, String eier);
    Optional<SoknadUnderArbeid> hentEttersendingMedTilknyttetBehandlingsId(String behandlingsId, String eier);
    List<SoknadUnderArbeid> hentAlleSoknaderUnderArbeidSiste15Dager();
    void oppdaterSoknadsdata(SoknadUnderArbeid soknadUnderArbeid, String eier) throws SamtidigOppdateringException;
    void oppdaterInnsendingStatus(SoknadUnderArbeid soknadUnderArbeid, String eier);
    void slettSoknad(SoknadUnderArbeid soknadUnderArbeid, String eier);
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/repository/soknadunderarbeid/SoknadUnderArbeidRepository.kt
package no.nav.sosialhjelp.soknad.repository.soknadunderarbeid

interface SoknadUnderArbeidRepository {
    fun opprettSoknad(soknadUnderArbeid: SoknadUnderArbeid, eier: String): Long?
    fun hentSoknad(soknadId: Long, eier: String): SoknadUnderArbeid?
    fun hentSoknad(behandlingsId: String?, eier: String): SoknadUnderArbeid
    fun hentSoknadNullable(behandlingsId: String?, eier: String): SoknadUnderArbeid?
    fun hentEttersendingMedTilknyttetBehandlingsId(tilknyttetBehandlingsId: String, eier: String): SoknadUnderArbeid?
    fun oppdaterSoknadsdata(soknadUnderArbeid: SoknadUnderArbeid, eier: String)
    fun oppdaterInnsendingStatus(soknadUnderArbeid: SoknadUnderArbeid, eier: String)
    fun slettSoknad(soknadUnderArbeid: SoknadUnderArbeid, eier: String)
}
